openedx/core/djangoapps/util/model_utils.py

```diff
@@ -0,0 +1,30 @@
+from django.db.models.fields import NOT_PROVIDED
+
+
+class Creator(object):
+    """
+    A placeholder class that provides a way to set the attribute on the model.
+    """
+    def __init__(self, field):
+        self.field = field
+
+    def __get__(self, obj, type=None):
+        if obj is None:
+            return self
+        return obj.__dict__[self.field.name]
+
+    def __set__(self, obj, value):
+        obj.__dict__[self.field.name] = self.field.to_python(value)
+
+
+class CreatorMixin(object):
+    """
+    Mixin class to provide SubfieldBase functionality to django fields.
+    See: https://docs.djangoproject.com/en/1.11/releases/1.8/#subfieldbase
+    """
+    def contribute_to_class(self, cls, name, private_only=False, virtual_only=NOT_PROVIDED):
```

Comment:
Why not specify `**kwargs` on this line instead of the full `private_only=False, virtual_only=NOT_PROVIDED` ?

Reply:
longer explanation:

Wouldn't the superclass already have these defaults specified, in which case they need not be set here?  Or are you intentionally setting them?

Reply:
Fixed.